src/edu/asu/plp/tool/prototype/Main.java

```diff
@@ -413,45 +463,843 @@ private ProjectExplorerTree createProjectTree()
+		grid.setPadding(new Insets(10, 10, 10, 10));
+		
+		Label projectName = new Label();
+		projectName.setText("Project Name: ");
+		projectName.setFont(Font.font("Arial", FontWeight.NORMAL, 16));
+		
+		TextField projTextField = new TextField();
+		projTextField.setText("Project Name");
+		projTextField.requestFocus();
+		projTextField.setPrefWidth(200);
+		
+		Label mainSourceFile = new Label();
+		mainSourceFile.setText("File Name: ");
+		mainSourceFile.setFont(Font.font("Arial", FontWeight.NORMAL, 16));
+		
+		TextField sourceFileField = new TextField();
+		projTextField.setPrefWidth(200);
+		
+		Label projectLocation = new Label();
+		projectLocation.setText("Location: ");
+		projectLocation.setFont(Font.font("Arial", FontWeight.NORMAL, 16));
+		
+		TextField projLocationField = new TextField();
+		projTextField.setPrefWidth(200);
+		
+		Button browseLocation = new Button();
+		browseLocation.setText("Browse");
+		browseLocation.setOnAction(new EventHandler<ActionEvent>() {
+			@Override
+			public void handle(ActionEvent e)
+			{
+				String chosenLocation = "";
+				FileChooser fileChooser = new FileChooser();
+				// fileChooser.setTitle("Choose Project Location");
+				DirectoryChooser directoryChooser = new DirectoryChooser();
+				//directoryChooser.getExtensionFilters().add(new
+				// FileChooser.ExtensionFilter("PLP files (*.plp)", "*.plp"));
+				// directoryChooser.setInitialDirectory(projTextField.getText());
+				directoryChooser.setTitle("Choose Project Location");
+				File file = directoryChooser.showDialog(null);
+				// If Cancel is chosen, throws a null pointer, needs to be fixed
+				chosenLocation = file.getAbsolutePath();
+				projLocationField.setText(chosenLocation);
+			}
+		});
+		
+		Label target = new Label();
+		target.setText("Targetted ISA: ");
+		target.setFont(Font.font("Arial", FontWeight.NORMAL, 16));
+		
+		ComboBox<String> isaType = new ComboBox<String>();
+		isaType.getItems().addAll("PLP", "MIPS");
+		isaType.setValue("PLP");
+		
+		Label version = new Label();
+		version.setText("Version: ");
+		version.setFont(Font.font("Arial", FontWeight.NORMAL, 16));
+		
+		grid.add(projectName, 0, 0);
+		grid.add(projTextField, 1, 0);
+		grid.add(mainSourceFile, 0, 1);
+		grid.add(sourceFileField, 1, 1);
+		grid.add(projectLocation, 0, 2);
+		grid.add(projLocationField, 1, 2);
+		grid.add(browseLocation, 2, 2);
+		grid.add(target, 0, 3);
+		grid.add(isaType, 1, 3);
+		grid.add(version, 0, 4);
+		
+		border.setCenter(grid);
+		
+		Button createProject = new Button("Create Project");
+		createProject.setOnAction(new EventHandler<ActionEvent>() {
+			@Override
+			public void handle(ActionEvent e)
+			{
+				String projectName;
+				String fileName;
+				String projectLocation;
+				Alert alert = new Alert(AlertType.INFORMATION);
+				projectName = projTextField.getText();
+				fileName = sourceFileField.getText();
+				projectLocation = projLocationField.getText();
+				if (projectName.equals(""))
```

Comment:
These checks should include a null-value check (just in case) and use trim() and isEmpty() instead of .equals(""). As so:
```
if (projectName == null || projectName.trim().isEmpty())
```